models/external-accounts.js

```diff
@@ -11,4 +11,33 @@ const addExternalAccountData = async (data) => {
   }
 };
 
-module.exports = { addExternalAccountData };
+const fetchExternalAccountData = async (query, param) => {
+  try {
+    const userExternalAccountData = [];
+
+    let externalAccountData;
+    externalAccountData = await externalAccountsModel.where("token", "==", param);
+
+    Object.keys(query).forEach((key) => {
+      if (key === "type") {
+        externalAccountData = externalAccountData.where(key, "==", query?.key);
+      }
+    });
+
+    externalAccountData = await externalAccountData.limit(1).get();
+
+    externalAccountData.forEach((data) => {
```

Comment:
```js
const querySnapshot = await externalAccountData.limit(1).get();
if (!querySnapshot.empty) {
  // do something with the first matching document
} else {
  // handle case where there are no matching documents
}

```

Reply:
Thanks for the snippet, made that change